synapse/handlers/message.py

```diff
@@ -553,21 +553,67 @@ def handle_new_client_event(
         event,
         context,
         ratelimit=True,
-        extra_users=[]
+        extra_users=[],
     ):
-        # We now need to go and hit out to wherever we need to hit out to.
-
-        # If we're a worker we need to hit out to the master.
-        if self.config.worker_app:
-            yield send_event_to_master(
-                self.http_client,
-                host=self.config.worker_replication_host,
-                port=self.config.worker_replication_http_port,
-                requester=requester,
-                event=event,
-                context=context,
+        """Processes a new event. This includes checking auth, persisting it,
+        notifying users, sending to remote servers, etc.
+
+        If called from a worker will hit out to the master process for final
+        processing.
+
+        Args:
+            requester (Requester)
+            event (FrozenEvent)
+            context (EventContext)
+            ratelimit (bool)
```

Comment:
we don't seem to do  much with this when we have a worker.

Reply:
Ah, and ditto for the `extra_users` param. I've added a commit to send the two params over replication.